src/main/java/de/tum/in/www1/artemis/service/TextExerciseFeedbackService.java

```diff
@@ -0,0 +1,156 @@
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.domain.enumeration.AssessmentType;
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+import de.tum.in.www1.artemis.domain.participation.Participation;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+import de.tum.in.www1.artemis.repository.StudentParticipationRepository;
+import de.tum.in.www1.artemis.service.connectors.athena.AthenaFeedbackSuggestionsService;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.InternalServerErrorException;
+import de.tum.in.www1.artemis.web.websocket.ResultWebsocketService;
+
+@Profile(PROFILE_CORE)
+@Service
+public class TextExerciseFeedbackService {
+
+    private static final Logger log = LoggerFactory.getLogger(TextExerciseFeedbackService.class);
+
+    public static final String NON_GRADED_FEEDBACK_SUGGESTION = "NonGradedFeedbackSuggestion:";
+
+    private final Optional<AthenaFeedbackSuggestionsService> athenaFeedbackSuggestionsService;
+
+    private final ResultWebsocketService resultWebsocketService;
+
+    private final SubmissionService submissionService;
+
+    private final ParticipationService participationService;
+
+    private final ResultService resultService;
+
+    private final ResultRepository resultRepository;
+
+    private final TextSubmissionService textSubmissionService;
+
+    public TextExerciseFeedbackService(Optional<AthenaFeedbackSuggestionsService> athenaFeedbackSuggestionsService, SubmissionService submissionService,
+            ResultService resultService, ResultRepository resultRepository, StudentParticipationRepository studentParticipationRepository,
+            ResultWebsocketService resultWebsocketService, ParticipationService participationService, TextSubmissionService textSubmissionService) {
+        this.athenaFeedbackSuggestionsService = athenaFeedbackSuggestionsService;
+        this.submissionService = submissionService;
+        this.resultService = resultService;
+        this.resultRepository = resultRepository;
+        this.resultWebsocketService = resultWebsocketService;
+        this.participationService = participationService;
+        this.textSubmissionService = textSubmissionService;
+    }
+
+    private void checkRateLimitOrThrow(StudentParticipation participation) {
+
+        List<Result> athenaResults = participation.getResults().stream().filter(result -> result.getAssessmentType() == AssessmentType.AUTOMATIC_ATHENA).toList();
+
+        long countOfAthenaResults = athenaResults.size();
+
+        if (countOfAthenaResults >= 10) {
+            throw new BadRequestAlertException("Maximum number of AI feedback requests reached.", "participation", "preconditions not met");
```

Comment:
Can this be localized on the client somehow? Also with the other errors
![image](https://github.com/user-attachments/assets/a69d51a7-8931-4b1a-b40c-bac9b13ad159)